internal/cmd/internal/olmv1/catalog_delete.go

```diff
@@ -0,0 +1,32 @@
+package olmv1
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/operator-framework/kubectl-operator/internal/cmd/internal/log"
+	v1action "github.com/operator-framework/kubectl-operator/internal/pkg/v1/action"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+)
+
+// NewCatalogDeleteCmd allows deleting a specified, existing catalog
+func NewCatalogDeleteCmd(cfg *action.Configuration) *cobra.Command {
+	i := v1action.NewCatalogDelete(cfg)
+	i.Logf = log.Printf
+
+	cmd := &cobra.Command{
+		Use:     "catalog <catalog_name>",
```

Comment:
It'd be useful to add a label filter or an `--all` option to use in case a catalog name isn't specified

Reply:
I agree, `--all` sounds good.
Regarding possible `selector` I think it's best if it's introduced separately, since it would make sense for multiple commands

Reply:
I added `--all` with deletions waits being done synchronously and using the same command-level timeout. I think that's fine now, but I'll add an issue to track potentially adding an async option to speed things up. This also kind of depends on how to handle waiting - https://github.com/operator-framework/kubectl-operator/pull/220#discussion_r1973356442